### DISCUSSION TITLE
Add a Roadmap/Gantt View to Projects (Beta) !!!!!!!!!!!!!!!

### DISCUSSION THREAD

**Discussion by MPRJ06** [community/community#6553] [Projects and Issues] [answered]
Please add a roadmap / Gantt chart view to the new Projects module.

**Comment by ynouri** [comment]
Worth pointing out that Notion has been doing an incredible work with their Timeline feature:
https://www.notion.so/guides/timeline-view-unlocks-high-output-planning-for-your-team

We are currently migrating to Github Issues but we have been forced to set up a (flimsy) integration Github --> Notion since our team is so reliant on the Notion Timeline view for planning and sequencing purposes.

2 important notes:
- We use the timeline feature for **"micro planning" ***within*** sprints**, which last anywhere between 1 and 3 weeks. We have observed that other products with a timeline / gantt views don't usually work very well at the day level (and are designed with long, multi-months roadmaps in mind). The Notion Timeline works splendidly at the day level, and we hope Github Timeline will do so as well!
- Another killer feature of the Notion Timeline is how easy it is to **drag and drop and resize an issue in the time dimension** (ie change start date, end date, length) directly by moving the issues visual blocks within the views. Planning is 100x easier this way than having to manually set a start date and end date in each individual tickets (as again, we have seen in other products offering a timeline / gantt view). We hope moving things around is as easy in Github Timeline!

**Reply by genidma** [reply]
> Worth pointing out that Notion has been doing an incredible work with their Timeline feature: https://www.notion.so/guides/timeline-view-unlocks-high-output-planning-for-your-team
> 
> We are currently migrating to Github Issues but we have been forced to set up a (flimsy) integration Github --> Notion since our team is so reliant on the Notion Timeline view for planning and sequencing purposes.
> 
> 2 important notes:
> 
> * We use the timeline feature for **"micro planning" _**within**_ sprints**, which last anywhere between 1 and 3 weeks. We have observed that other products with a timeline / gantt views don't usually work very well at the day level (and are designed with long, multi-months roadmaps in mind). The Notion Timeline works splendidly at the day level, and we hope Github Timeline will do so as well!
> * Another killer feature of the Notion Timeline is how easy it is to **drag and drop and resize an issue in the time dimension** (ie change start date, end date, length) directly by moving the issues visual blocks within the views. Planning is 100x easier this way than having to manually set a start date and end date in each individual tickets (as again, we have seen in other products offering a timeline / gantt view). We hope moving things around is as easy in Github Timeline!

Well, if it rivals or is better than the notion like view, then I'd definitely pay for this service. Thx

**Comment by erikr** [comment]
Timelines are on their road map, per https://github.com/features/issues/, but no idea when:

![CleanShot 2021-10-22 at 11 19 06](https://user-images.githubusercontent.com/1253422/138480568-e1bc2473-9673-465e-a631-c492058a9c30.png)

https://linear.app has this now and it looks beautiful:
![CleanShot 2021-10-22 at 11 19 27](https://user-images.githubusercontent.com/1253422/138480629-892f56e5-7112-4d76-a1b3-a87663363fcc.png)

Is there an ETA of this feature? Similar to @ynouri, my team is also reliant on Notion timeline view. We may switch to Linear to do everything in one place.

**Reply by erikr** [reply]
Hi @azenMatt, any update on this? Last few weeks of managing boards and timelines in two different platforms has been extremely painful. Thanks so much!

**Reply by azenMatt** [reply]
Hi @erikr - I don't have a timeline to share on this just yet, but it is still in our plans to add this 😄

**Reply by ynouri** [reply]
@azenMatt, kind remark on the announcement of the timeline feature:

For us, the "timelines (coming soon)" part of the [announcement page](https://github.com/features/issues) was _the_ major factor in our decision to jump ships and try out Github Issues back in October. I am sure it was a key factor for many other teams, and that this mention didn't make its way in the announcement by chance.

Obviously "soon" is loosely defined, but I am sure that you understand that after a few months this starts looking more and more like a bait-and-switch strategy to the eyes of stakeholders who are not as much committed to Github Issues. (Yes, the 3 engineers who invested time and effort in migrating tickets, setting workflows, boards, giving feedback, and training the rest of the team have no desire to get another tool as they are otherwise very happy with the product. But the executive who didn't go through that pain and needs that timeline feature badly, _he_ could pull the trigger and make the entire team switch to Linear any moment.)

It would be really great to at least get a rough estimate (Q1? Q2? ...), or a sense of the level of commitment / relative priority in your backlog.

![image](https://user-images.githubusercontent.com/1789921/145897380-8a9a9e07-57c7-489d-b20c-1146c608c3a8.png)

**Reply by erikr** [reply]
I completely agree with @ynouri. Going to try Linear and see well it imports issues from GH.

**Reply by azenMatt** [reply]
Hi @ynouri and @erikr - I appreciate your feedback on this, the timeline is still a feature that we will be bringing to the new projects but I don't have a timeline on when that would be to share with you at the moment. As soon as we have something to share, I will make sure this gets updated.

**Reply by erikr** [reply]
As your team is thinking about what to prioritize in the new year, might we gently suggest this as a super valuable and awesome feature? :) Thanks!

**Reply by ynouri** [reply]
Hi @azenMatt , as of June 2022 the "timelines (coming soon)" mention still appears on [your product page](https://github.com/features/issues). This was announced in October 2021. I totally understand you guys are super busy, and it's OK to deprioritize features, but I think it's a lack of respect for the users to keep announcing a major feature that is not coming. Or replace "coming soon" with "in 1 year or more". I love Github and I'm usually championing your products, but here you guys are making yourself really hard to defend 😕

**Reply by erikr** [reply]
Their [public road map](https://github.com/orgs/github/projects/4247/views/1?filterQuery=timeline) shows "timeline" still labeled as `Future` (https://github.com/github/roadmap/issues/278). So sounds like it is >1 year away. Painful!

My team couldn't wait any longer, so we now track higher level OKRs in Notion. We only use GitHub for more granular issue tracking since it is still so bare bones and especially lacking this timeline feature.

**Reply by alper** [reply]
I think Notion+Github is a fine combination.

**Reply by azenMatt** [reply]
Hi - thanks for all the feedback on this. I appreciate that this can be frustrating! The timeline layout is still something we strong believe is a great enhancement to the product and are committed to building. However I don't have a date I can share with you at this point on when it will be ready. We will keep you posted as we get closer.

**Reply by joshjohanning** [reply]
[Timeline in projects](https://github.com/github/roadmap/issues/278) was added to Q4 2022 recently

**Reply by erikr** [reply]
After months of no update, my org moved all task tracking and project management to Notion.

**Reply by ynouri** [reply]
We moved to Jira - their [Roadmap feature](https://www.atlassian.com/software/jira/guides/roadmaps/basic-roadmaps) is quite good.

**Comment by Kleywalker** [comment]
A timeline view will be great! 😊 I hope this feature will be coming very soon! 👍 Are there any news on this yet?

**Comment by andy-jacobs** [comment]
I would love to have this feature too!

**Reply by MPRJ06** [reply]
No point in waiting. Try switching to Shortcut (formerly Clubhouse - not to be confused with the social app). Has everything you need and is way easier to use than GitHub or Jira.

**Reply by andy-jacobs** [reply]
Got it, thanks!

**Reply by erikr** [reply]
https://shortcut.com/

I wonder what GitHub uses for their internal project management and time tracking.

**Comment by ceciliacfx** [comment]
And in the meantime, while GitHub works on building the roadmap / gantt capability...could we please at least have the option to export the data so we can use a different tool?

**Comment by erikr** [comment]
Hey @azenMatt, three months ago you did not have a timeline for us on this oft-requested functionality. Any news since then? Thanks.

**Reply by azenMatt** [reply]
Hi @erikr - I do not. I'l make sure to keep you posted when we have more information. 

We do have lots of the top requested items coming soon, including https://github.com/github/feedback/discussions/5381 which is currently the most requested item for the new projects. I appreciate your patience waiting for this future - I know I am excited for it as well!

**Reply by d3chapma** [reply]
Could be wrong, but looks like this is scheduled for this quarter: https://github.com/github/roadmap/issues/483

Maybe that is talking about a different timelines feature?

**Reply by erikr** [reply]
I am not clear what functionality is described in https://github.com/github/roadmap/issues/483 but the topic of this Discussion is the timeline functionality best showcased by https://linear.app/docs/roadmap-doc.

**Comment by d3chapma** [comment]
I see now that the issue I was pointing to is not the same feature. Rather it is [this one](https://github.com/github/roadmap/issues/278) that is tracking the timeline feature: 

Based on where it is in the [public roadmap](https://github.com/orgs/github/projects/4247/views/7), it would seem that it _might_ be available by end of year...

**Comment by skullface** [comment]
this should be prioritized on the planning roadmap or removed from the marketing page 🙇‍♀️

**Comment by erikr** [comment]
My team is switching off of GitHub for work tracking because of the lack of this feature.

**Comment by alexanderniebuhr** [comment]
hey @azenMatt, @2percentsilk can you share a top secret ETA for the beta date for the timeline feature (https://github.com/github/roadmap/issues/278)?

**Comment by rileybroughten** [comment, marked as answer]
👋 Hello all! Please check out the [announcement](https://github.blog/2022-11-09-everything-new-from-github-universe-2022/#plan-and-track-your-work-in-the-same-place-you-build) of Roadmap (and Tasklists) from Universe last month, and check out the [waitlist signup](https://github.com/features/issues/signup) to get your organization(s) added! 👍
